app/src/main/java/net/osmtracker/gpx/ExportToTempFileTask.java

```diff
@@ -31,6 +34,18 @@ public ExportToTempFileTask(Context context, long trackId) {
 			throw new IllegalStateException("Could not create temporary file", ioe);
 		}
 	}
+
+	public ExportToTempFileTask(Context context, long trackId, ContentResolver contentResolver) {
```

Comment:
Instead of creating a new method, try to adapt the existing one and extend it to include the name of file temp file in both functionalities (share and upload to osm)